app/src/main/java/com/github/studydistractor/sdp/eventHistory/EventHistoryServiceFirebase.kt

```diff
@@ -0,0 +1,128 @@
+package com.github.studydistractor.sdp.eventHistory
+
+import android.util.Log
+import androidx.compose.runtime.mutableStateListOf
+import com.github.studydistractor.sdp.data.Event
+import com.github.studydistractor.sdp.data.EventParticipants
+import com.github.studydistractor.sdp.data.FirebaseEvent
+import com.github.studydistractor.sdp.data.FirebaseEventParticipants
+import com.google.android.gms.tasks.Task
+import com.google.android.gms.tasks.Tasks
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.database.DataSnapshot
+import com.google.firebase.database.DatabaseError
+import com.google.firebase.database.DatabaseReference
+import com.google.firebase.database.FirebaseDatabase
+import com.google.firebase.database.ValueEventListener
+
+class EventHistoryServiceFirebase: EventHistoryModel {
+    private val eventPaths = "Events"
+    private val eventDatabaseRef: DatabaseReference = FirebaseDatabase.getInstance().getReference(eventPaths)
+    private val eventParticipantsPaths = "EventParticipants"
+    private val eventParticipantsDatabaseRef: DatabaseReference = FirebaseDatabase.getInstance().getReference(eventParticipantsPaths)
+    private var auth : FirebaseAuth = FirebaseAuth.getInstance()
+
+    private var onEventChange : (List<Event>) -> Unit = {}
+    private var onEventParticipantsChange : (List<EventParticipants>) -> Unit = {}
+
+    val firebaseEvents = mutableStateListOf<FirebaseEvent>()
+    val firebaseEventParticipants = mutableStateListOf<FirebaseEventParticipants>()
+
+    private val eventListener = object : ValueEventListener {
+        override fun onDataChange(snapshot: DataSnapshot) {
+            firebaseEvents.clear()
+            for (event in snapshot.children) {
+                val eventItem = event.getValue(FirebaseEvent::class.java)
+                if (eventItem != null) {
+                    firebaseEvents.add(eventItem)
+                }
+            }
+
+            getEvents().continueWith{
+                t -> onEventChange(t.result)
```

Comment:
You can create a list in this method and call onEventChange() with it no need to use the mutableStateList.